runtime/src/accounts_db.rs

```diff
@@ -1047,46 +1062,118 @@ impl AccountsDB {
         datapoint_info!("accounts_db-stores", ("total_count", total_count, i64));
     }
 
-    pub fn verify_bank_hash(
+    fn accumulate_account_hashes(mut hashes: Vec<(Pubkey, Hash)>) -> Hash {
+        let mut sort = Measure::start("sort");
+        hashes.par_sort_by(|a, b| a.0.cmp(&b.0));
+        sort.stop();
+        let mut hash_time = Measure::start("hash");
+        let hashes: Vec<_> = hashes.chunks(16).collect();
+        let hashes: Vec<_> = hashes
+            .par_iter()
+            .map(|k| {
+                let mut hasher = Hasher::default();
+                for v in k.iter() {
+                    hasher.hash(v.1.as_ref());
+                }
+                hasher.result()
+            })
+            .collect();
+        let mut hasher = Hasher::default();
+        for hash in &hashes {
```

Comment:
This part might be our next bottle neck as we go higher TPS as hardware improves. For such cases, we need hard-fork to change this hashing algorithm, right?

Btw, a bit dump question, merkletree didn't work for accounts hash?

Reply:
merkle tree was a bit too slow.  @sakridge and I where talking about making this a full tree base 16, and computing the full hash asynchronously into the state.  So any time bank_height (not slot) % Period == 0, we would compute the full account hash for that bank and then make it available in in the state of bank_height + Period

Reply:
Oh I see! thanks for explanation!